src/robot.ts

```diff
@@ -149,19 +160,19 @@ class Robot {
    * @returns {Promise<github>} - An authenticated GitHub API client
    * @private
    */
-  async auth (id, log = this.log) {
-    const github = new GitHubApi({
-      debug: process.env.LOG_LEVEL === 'trace',
+  public async auth (id?: number, log = this.log) {
```

Comment:
Can you add the return type here as well?  Looks like `Promise<OctokitWithPagination>`.